include/dali/core/util.h

```diff
@@ -84,6 +84,23 @@ constexpr Value align_up(Value v, Alignment a) {
   return v + ((a - 1) & -v);
 }
 
+template <typename Value, typename Alignment>
+DALI_HOST_DEV
+constexpr Value align_reminder(Value v, Alignment a) {
+  return v & (a - 1);
+}
+
+template <typename Value, typename Alignment>
+DALI_HOST_DEV
+constexpr Value align_down(Value v, Alignment a) {
+  return v & ~(a - 1);
+}
+
+template <typename Value, typename Offset, typename Alignment>
+DALI_HOST_DEV
+constexpr Value align_up_offset(Value v, Offset o, Alignment a) {
+  return align_up(v + align_reminder(o, a), a);
```

Comment:
It seems rather convoluted for a general purpose utility. Just by looking at it (without seeing the usage) it's hard to guess what it might be useful for.

Reply:
Any idea where to put it? I use it in at least 2 different places.

Reply:
After all we can go with the bellow approach and remove`align_up_offset`.

Reply:
Done